osdep/threads.h

```diff
@@ -9,4 +9,8 @@
 #include "threads-posix.h"
 #endif
 
+#if (__STDC_VERSION__ < 202311L) && !defined(thread_local)
+#define thread_local _Thread_local
+#endif
```

Comment:
osdep/compiler.h

Reply:
I considered that, but I'd expect to get thread_local from `osdep/threads.h` (similar to how it's defined in `<threads.h>` in standard headers).

Also the only usage of this already includes `osdep/threads.h` but doesn't include `osdep/compiler.h` directly. So it might be easy to mess up the include whereas including `osdep/threads.h` should be natural for code using `thread_local`.

Reply:
> I considered that, but I'd expect to get thread_local from osdep/threads.h (similar to how it's defined in <threads.h> in standard headers).

Then include compiler.h in threads.h, it's pointless to duplicate and scatter compatibility definitions when we have dedicated header for this purpose.

Reply:
Also this will not work for C++ where `_Thread_local` is not defined and `thread_local` is native.